flink-cdc-connect/flink-cdc-source-connectors/flink-connector-tidb-cdc/src/main/java/org/apache/flink/cdc/connectors/tidb/TDBSourceOptions.java

```diff
@@ -82,10 +90,13 @@ private TDBSourceOptions() {}
                     .withDescription("TiKV GRPC batch scan concurrency");
 
     public static TiConfiguration getTiConfiguration(
-            final String pdAddrsStr, final Map<String, String> options) {
+            final String pdAddrsStr, final String hostMapping, final Map<String, String> options) {
         final Configuration configuration = Configuration.fromMap(options);
 
         final TiConfiguration tiConf = TiConfiguration.createDefault(pdAddrsStr);
+        if (StringUtils.isNotBlank(hostMapping)) {
```

Comment:
Could we use Flink's `Preconditions` check utils instead of introducing external dependency ?

Reply:
Optional.of(new UriHostMapping(hostMapping)).ifPresent(tiConf::setHostMapping);